src/dist/replication/test/replica_test/unit_test/replica_disk_test.cpp

```diff
@@ -0,0 +1,167 @@
+
+namespace dsn {
+namespace replication {
+
+class replica_disk_test : public replica_test_base
+{
+public:
+    int dir_nodes_count = 5;
+    int primary_count_for_disk = 1;
+    int secondary_count_for_disk = 2;
+    app_id app_id_1 = 1;
+    app_id app_id_2 = 2;
+    dsn::app_info app_info_1;
+    dsn::app_info app_info_2;
+
+public:
+    void SetUp() override
+    {
+        generate_mock_app_info();
+        generate_mock_dir_nodes(dir_nodes_count);
+        stub->generate_replicas_base_dir_nodes_for_app(
+            app_info_1, primary_count_for_disk, secondary_count_for_disk);
+        secondary_count_for_disk = 0;
+        stub->generate_replicas_base_dir_nodes_for_app(
+            app_info_2, primary_count_for_disk, secondary_count_for_disk);
+        stub->on_disk_stat();
+    }
+
+    std::vector<std::shared_ptr<dir_node>> get_fs_manager_nodes()
+    {
+        return stub->_fs_manager._dir_nodes;
+    }
+
+private:
+    void generate_mock_app_info()
+    {
+        app_info_1.app_id = 1;
+        app_info_1.app_name = "disk_test_1";
+        app_info_1.app_type = "replica";
+        app_info_1.is_stateful = true;
+        app_info_1.max_replica_count = 3;
+        app_info_1.partition_count = 8;
+
+        app_info_2.app_id = 2;
+        app_info_2.app_name = "disk_test_2";
+        app_info_2.app_type = "replica";
+        app_info_2.is_stateful = true;
+        app_info_2.max_replica_count = 3;
+        app_info_2.partition_count = 16;
+    }
+
+    void generate_mock_dir_nodes(int num)
+    {
+        int64_t disk_capacity_mb = num * 100;
+        while (num > 0) {
+            int64_t disk_available_mb = num * 50;
+            int disk_available_ratio =
+                static_cast<int>(std::round((double)100 * disk_available_mb / disk_capacity_mb));
+            // create one mock dir_node and make sure disk_capacity_mb_ > disk_available_mb_
+            dir_node *node_disk = new dir_node("tag_" + std::to_string(num),
+                                               "full_dir_" + std::to_string(num),
+                                               disk_capacity_mb,
+                                               disk_available_mb,
+                                               disk_available_ratio);
+            int replica_index = 0;
+            int disk_holding_replica_count = primary_count_for_disk + secondary_count_for_disk;
+            while (disk_holding_replica_count-- > 0) {
+                node_disk->holding_replicas[app_id_1].emplace(gpid(app_id_1, replica_index++));
+            }
+
+            replica_index = 0;
+            disk_holding_replica_count = primary_count_for_disk;
+            while (disk_holding_replica_count-- > 0) {
+                node_disk->holding_replicas[app_id_2].emplace(gpid(app_id_2, replica_index++));
+            }
+
+            stub->_fs_manager._dir_nodes.emplace_back(node_disk);
+            num--;
+        }
+    }
+};
+
+TEST_F(replica_disk_test, on_query_disk_info)
+{
```

Comment:
Could you please refactor this unit test? This test contains too many test cases and has many duplicated code.
You can write test in different TEST_F, such as `TEST_F(replica_disk_test, query_app_not_existed)`, `TEST_F(replica_disk_test, query_all_app)`. 
Besides, you can also define a test_struct to reduce duplicated code.
You can see more in [http_server_test.cpp](https://github.com/XiaoMi/rdsn/blob/master/src/dist/http/test/http_server_test.cpp)